wpimath/src/main/java/edu/wpi/first/math/estimator/KalmanFilter.java

```diff
@@ -197,9 +209,25 @@ public void predict(Matrix<Inputs, N1> u, double dtSeconds) {
    */
   @SuppressWarnings("ParameterName")
   public void correct(Matrix<Inputs, N1> u, Matrix<Outputs, N1> y) {
-    final var C = m_plant.getC();
     final var D = m_plant.getD();
+
+    @SuppressWarnings("LocalVariableName")
+    // S = CPCᵀ + R
+    var S = m_C.times(m_P).times(m_C.transpose()).plus(m_discR);
```

Comment:
m_discR needs to be rediscretized. https://github.com/wpilibsuite/allwpilib/blob/main/wpimath/src/main/java/edu/wpi/first/math/estimator/ExtendedKalmanFilter.java shows what this class should be doing in general, except for the Jacobian evaluations.